test/e2e/utils.go

```diff
@@ -353,6 +354,21 @@ func waitForDeploymentStatusConditions(ctx context.Context, cl client.Client, ti
 	})
 }
 
+func waitForDaemonsetUpToDate(ctx context.Context, cl client.Client, timeout time.Duration, name, ns string, numberUpToDateAndAvailable int32) error {
+	nsName := types.NamespacedName{
+		Namespace: ns,
+		Name:      name,
+	}
+	return wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		daemonset := &appsv1.DaemonSet{}
+		if err := cl.Get(ctx, nsName, daemonset); err != nil {
+			return false, nil
+		}
+		return daemonset.Status.UpdatedNumberScheduled == numberUpToDateAndAvailable &&
```

Comment:
Is it necessary to check both or just number of available?

Reply:
my reasoning behind checking both was to make sure if the upgrade includes an envoy bump the daemonset is updated to that version and the check won't return early etc.